db/data/20231219120311_update_hesa_trainees_with_lead_school_urn.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class UpdateHesaTraineesWithLeadSchoolUrn < ActiveRecord::Migration[7.1]
+  def up
+    trainees.each do |trainee|
+      lead_shool = find_lead_school(trainee)
+
+      next unless lead_shool
+
+      trainee.update(lead_shool: lead_shool, lead_school_not_applicable: false)
+    end
+  end
+
+  def down
+    raise ActiveRecord::IrreversibleMigration
+  end
+
+private
+
+  def trainees
+    Trainee.left_joins(:hesa_students)
+           .where(lead_school_id: nil)
+           .where.not(hesa_students: { lead_school_urn: %w[900000 900010 900020 900030] })
+           .where.not(hesa_students: { lead_school_urn: nil })
+           .distinct
+  end
```

Comment:
FYI: this returns about ~70 trainee records

Reply:
presumably get the same number if we add a filter `.where(lead_school_not_applicable: true)`?

Reply:
actually i get
```
irb> Trainee.left_joins(:hesa_students).where(lead_school_id: nil, lead_school_not_applicable: true).where.not(hesa_students: { lead_school_urn: %w[900000 900010 900020 900030] }).where.not(hesa_students
: { lead_school_urn: nil }).distinct.count
=> 33
```

so probably we should include the filter for `lead_school_not_applicable`